homeassistant/components/history/__init__.py

```diff
@@ -64,6 +67,7 @@ def get_significant_states(
         query = session.query(States).filter(
             (
                 States.domain.in_(SIGNIFICANT_DOMAINS)
+                | (significant_changes_only and States.domain.is_("device_tracker"))
                 | (States.last_changed == States.last_updated)
```

Comment:
This is too complicated and not easy to built on top.  When `significant_changes_only` is set to True, we should just drop this whole filter. That's it.

Reply:
All other changes need to be reverted.

Reply:
Done. @Hoytron you have to invert the logic of `significant_changes_only`, since you now want to receive all changes and not only the significant ones.